lib/rspec/core/memoized_helpers.rb

```diff
@@ -232,6 +232,9 @@ def let(name, &block)
           # We have to pass the block directly to `define_method` to
           # allow it to use method constructs like `super` and `return`.
           raise "#let or #subject called without a block" if block.nil?
+          if method_defined?(name) && instance_method(name).owner == self
+            RSpec.warning "method :#{name} defined multiple times through let in the same context"
```

Comment:
This warning string is accurate for your case and the specs you wrote below, but there are cases where it would be emitted and would not be accurate:

``` ruby
RSpec.describe do
  def foo; end
  let(:foo) { }
end
```

In such a situation, `let(:foo)` is stomping `def foo` but the warning describes both as `let`.  I also think that this warning could be a bit clearer in general -- it says there are multiple `let(#{name})` but doesn't explain why that is problematic.  How about this instead?

``` ruby
RSpec.warning "`#{self.name}##{name}` is being redefined through the `let(:#{name})` " \
              "definition at #{RSpec::CallerFilter.first_non_rspec_line}. The original " \
              "definition will never be used and can be removed", :call_site => nil
```

...which would look like:

```
WARNING: `RSpec::ExampleGroups::Anonymous#foo` is being redefined through the `let(:foo)` definition at /Users/myron/code/rspec-dev/repos/rspec-core/spec/rspec/core/memoized_helpers_spec.rb:436:in `block (3 levels) in <module:Core>'. The original definition will never be used and can be removed.
```

I think that's more clear.  WDYT?